### ISSUE TITLE
Help wanted?

### ISSUE THREAD

**Issue by aisflat439** [leighs-hammer/shopify-app-boilerplate-nextjs-redux-nosql#13]
I just pulled this down, followed the readme and was pleasantly surprised that it actually worked! Is there anything you need done as I play around with this?

**Comment by leighs-hammer** [comment]
Hey @aisflat439 there are a few odditties I haven't patched backwards from actual live builds. 

I will try dig them out and pass them over ( this know webhook validation was failing and a few other production deployment changes ) 

I have been a little slack getting the rest in sorry for that, I was gonna divert my attention to rewriting it for the new auth flow that has been proposed and ripping out redux, or revising it for the new jamstack focus by the team at vercel. 

Hit me up if you hit any road blocks I have probably fixed them in other projects based off this :) 

As far as help, I was as I said gonna rebuild it somewhat and would love any input this is just out there for people to use if it helps them as an alternative to the cli. ( I was planning on adding that as well at some point )

Glad you got it up and running! 

Cheers
Leigh B

**Comment by aisflat439** [comment]
Indeed! I think pulling out redux might be smart. I quite like redux but it's probably easier for people looking to get started not to have to think about it. As I work today I'll see if I should make a PR for any of the things I find today. Leaning into Vercel instead of Koa makes a ton of sense to me personally. It's much more straightforward. Thanks for this starting point and let me know if I can do anything.

**Comment by aisflat439** [comment]
Hey Leigh! Thanks for merging in my PR. I'll be happy to try and contribute some more as I get time. How are you usually using the `useQuery` hook. I tried to drop it in on dashboard just to start to play with how the query is set up.

```js
const gql = `query {
  customers(first: 10) {
    edges{
      node {
        firstName
        lastName
        email
      }
    }
  }
}`

const Dashboard = () => {
  const router = useRouter()
  const { data, fetching, error } = useQuery(gql, {}, {})
```
on page load, fetching is `fasle` which makes sense. then the query fires, we switch to `true` until we have data, then back to false. How do you not fire `useQuery` on load? I sorta expected something like

```js
const {runQuery} = useQuery()

const someHanlder = () => {
 const { data, fetching } =  runQuery(gql)
}
```
Or are you cautious to ensure that you only call the query inside components. 

If you have a trivial example of how you're using it that would be super. I'll be happy to write it up and add it to the Wiki as well. Thanks!

**Comment by leighs-hammer** [comment]
@aisflat439 oh god is that still in there! I actually did away with writing my own contexts and query runners and replaced it with react query by Tanner Linsey which is a phenomenal performance and feature bump: 

https://github.com/tannerlinsley/react-query

I would suggest installing this and then the use would be something like: 

```
import { useQuery } from 'react-query'
import axios from 'axios'
import { useSelector, useDispatch } from 'react-redux'


const SOME_GQL_CODE = `{
  shop {
    name
  }
}`

const SomeComponent = () => {

  // this is available in redux but you basically just need the myshopify domain for the query middleware/proxy to work. 

  // required
  const permanentDomain: string | false = useSelector(state => state.shop.domain)

  // optional
  const variables = {
    id: 'these may be needed for your query code'
  }


  // A function that react query uses to get data
  const getSomeData = async () => {
    const response = await axios.post(`/api/query`, {
      gql: SOME_GQL_CODE,
      shop: permanentDomain,
      variables: variables,
    })
    return response.data
  }
  
  const {data} = useQuery([`task-${id}`, permanentDomain], getSomeData)
  // optionally if you want to add options such as a refetch interval or stale time the docs are available in React querys repo
  // const {data} = useQuery([`task-${id}`, permanentDomain], getSomeData, {refetchInterval: 4000})


  // handle undefined or loading ( refer to react queries docs)
  if(!data) { return null}

  return (
    <div>
      <pre>{JSON.stringify(data)}</pre>
    </div>
  )
}

return default SomeComponent
```

This is a way more efficient method over custom hooks. Also allows for phenomenal caching and refetch out the box, eg if you needed to poll for changes every x seconds etc. 

That runQuery you referenced was a first attempt POC of using custom hooks to interact then I realized it was just over engineering. 

BTW: Are you on the shopify partners slack?

Cheers 
Leigh B

**Comment by leighs-hammer** [comment]
As a side note: 
[https://github.com/leighs-hammer/shopify-app-boilerplate-nextjs-redux-nosql/blob/master/pages/api/query.ts](url)

Does all the query leg work, so from the react side you can really use any system to make a POST request with the required params to then pass along and return from Shopify.

If you only need a single call you could simply use axios or even fetch to call that endpoint :)

**Comment by aisflat439** [comment]
NIce! One of the apps at work uses `react-query` but I haven't had a chance to use it. I saw a talk by Tanner not long ago and it seems pretty slick. I ended up doing exactly what you said, just use `/pages/api`, which is just as well. Thanks for looping me in about what was going on with `runQuery`. I agree that it will be easier to manage by leveraging something else.

I'm not on the Shopify Partners Slack. Are there some good channels there?

**Comment by leighs-hammer** [comment]
There are like appdev and tech channels pretty good to get cross partner comms, discussion and help. The partner ecosystem is a friendly place ful of loafs of helpful folk :) worth joining for sure

**Comment by aisflat439** [comment]
I'll join and say hi!
